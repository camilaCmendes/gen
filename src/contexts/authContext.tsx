import { ReactNode, createContext, useEffect, useState } from "react";
import { UserDTO } from "../dtos/userDTO";
import { storageUserListFind } from "../storages/storageUserList";
import {
  storageUserGet,
  storageUserRemove,
  storageUserSave,
  storageUserUpdate,
} from "../storages/storageUser";
import { useNavigate } from "react-router-dom";

export type AuthContextDataProps = {
  user: UserDTO;
  login: (email: string, password: string) => boolean;
  logout: () => void;
  changePassword: (user: UserDTO, newPassword: string) => void;
};

type AuthContextProviderProps = {
  children: ReactNode;
};

export const AuthContext = createContext<AuthContextDataProps>(
  {} as AuthContextDataProps
);

export const AuthContextProvider: React.FC<AuthContextProviderProps> = ({
  children,
}) => {
  const [user, setUser] = useState({} as UserDTO);
  const navigate = useNavigate();

  const login = (email: string, password: string) => {
    const userFound = storageUserListFind(email, password);
    if ("email" in userFound) {
      setUser(userFound);
      storageUserSave(userFound);
      return true;
    }
    return false;
  };

  const logout = () => {
    storageUserRemove();
    setUser({} as UserDTO);
  };

  const changePassword = (user: UserDTO, newPassword: string) => {
    storageUserUpdate({ ...user, password: newPassword });
    setUser({ ...user, password: newPassword });
  };

  useEffect(() => {
    const userIsLogged = storageUserGet();

    if (userIsLogged && userIsLogged.email) {
      console.log("ENTROU");
      setUser(userIsLogged);
    } else {
      navigate("/login");
    }
  }, []);

  return (
    <AuthContext.Provider value={{ user, login, logout, changePassword }}>
      {children}
    </AuthContext.Provider>
  );
};
