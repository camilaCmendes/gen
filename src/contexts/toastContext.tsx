import React, { createContext, useState, ReactNode } from "react";
import { Snackbar, Alert } from "@mui/material";

type ToastContextType = {
  showToast: (
    message: string,
    severity?: "error" | "warning" | "info" | "success"
  ) => void;
};

const ToastContext = createContext<ToastContextType>({} as ToastContextType);

export const ToastProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState("");
  const [severity, setSeverity] = useState<
    "error" | "warning" | "info" | "success"
  >("success");

  const showToast = (
    message: string,
    severity: "error" | "warning" | "info" | "success" = "success"
  ) => {
    setMessage(message);
    setSeverity(severity);
    setOpen(true);
  };

  const handleClose = (
    event?: React.SyntheticEvent | Event,
    reason?: string
  ) => {
    if (reason === "clickaway") {
      return;
    }
    setOpen(false);
  };

  return (
    <ToastContext.Provider value={{ showToast }}>
      {children}
      <Snackbar open={open} autoHideDuration={4000} onClose={handleClose}>
        <Alert
          onClose={handleClose}
          severity={severity}
          variant="filled"
          sx={{ width: "100%" }}
        >
          {message}
        </Alert>
      </Snackbar>
    </ToastContext.Provider>
  );
};

export default ToastContext;
