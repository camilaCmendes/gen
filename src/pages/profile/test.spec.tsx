import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Profile } from ".";
import { AuthContextProvider } from "../../contexts/authContext";
import { ToastProvider } from "../../contexts/toastContext";

const mockUser = {
  id: "4",
  userAccess: "ADMIN",
  name: "Leonardo Marzeuski",
  role: "Analista de Sistemas",
  team: "AnÃ¡lise",
  email: "leonardo@email.com",
  password: "123456",
};

jest.mock("../../hooks/useAuth", () => ({
  useAuth: () => ({
    user: mockUser,
    logout: jest.fn(),
    changePassword: jest.fn(),
    login: jest.fn(),
  }),
}));

jest.mock("../../hooks/useToast", () => ({
  useToast: () => ({
    showToast: jest.fn(),
  }),
}));

describe("Profile", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <Router>
        <AuthContextProvider>
          <ToastProvider>
            <Profile />
          </ToastProvider>
        </AuthContextProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should display user data correctly", () => {
    render(
      <Router>
        <AuthContextProvider>
          <Profile />
        </AuthContextProvider>
      </Router>
    );

    expect(screen.getByText(mockUser.name)).toBeInTheDocument();
    expect(
      screen.getByText("Especialidade: " + mockUser.role)
    ).toBeInTheDocument();
    expect(screen.getByText("Time: " + mockUser.team)).toBeInTheDocument();
  });
});
