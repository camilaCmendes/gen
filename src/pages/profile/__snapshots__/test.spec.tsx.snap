// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-skrp9k"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1diinbu-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1nits8g-MuiTypography-root"
        >
          Leonardo Marzeuski
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-183vntp-MuiTypography-root"
        >
          Especialidade: Analista de Sistemas
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-183vntp-MuiTypography-root"
        >
          Time: An√°lise
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-fm6yoz-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Senha Antiga
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-hwkq3c-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id=":r0:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-fm6yoz-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            Nova Senha
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-hwkq3c-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id=":r1:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-11p3wlv-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Atualizar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
